### ISSUE TITLE
Start using our self-hosted Sentry to catch the app crashes

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdguardForAndroid#1908]
## User consent

We need to extend the Terms step and add a new checkbox there:
`Allow AdGuard to send automatic crash reports. <a href="privacy policy link">Privacy Policy</a>.`

In the stable build, this checkbox should be unchecked by default.
In the beta&nightly builds, this checkbox should be checked.

## What to capture

The only thing we need to capture is the app crashes. So, extend the default unhandled exceptions handler. One more thing is the native crashes information acquired from CoreLibs (needs to be discussed with @sfionov).

Make sure that the report does not contain any personal information (or anything that can be used to identify the user).

Also, prepare the list of what's being sent. We will need it for https://github.com/AdguardTeam/LegalDocs/issues/14

## Sentry configuration

Here's how to use the Android version of Sentry:
https://docs.sentry.io/clients/java/modules/android/

Our self-hosted sentry instance is located on sentry.adguard.com.

**Comment by Revertron** [comment]
1. From the mentioned docs:
> An UncaughtExceptionHandler is configured so that crash events will be stored to disk and sent the next time the application is run.
This means that our UncaughtExceptionHandler  will be overridden. Do we need it?
2. Sentry relies on "breadcrumbs" (the log entries that we need to add everywhere). Do we need it?
3. If AdGuard will crash in native code, the sentry will not collect logcat and will not send it to us. But we need it.
4. For readability of the logs we need to not strip file names and line numbers. It will severely worsen obfuscation.

Bottom line, it may be useful for click-tracking in some marketing apps, but I don't see how it can help us in (native) logs collection.

May be, we need to extend our unhandled exceptions handler manually, and send crash-dumps manually too?

**Comment by ameshkov** [comment]
> This means that our UncaughtExceptionHandler will be overridden. Do we need it?

I'd prefer to not override it and use our own handler to report.

> 2. Sentry relies on "breadcrumbs" (the log entries that we need to add everywhere). Do we need it?

Why? The only thing we need is crash reports.

> 3. If AdGuard will crash in native code, the sentry will not collect logcat and will not send it to us. But we need it.

This is about to get resolved, don't worry.

> May be, we need to extend our unhandled exceptions handler manually, and send crash-dumps manually too?

This is exactly what we need. Manually -- but to our self-hosted Sentry instance so that it could take care of:

* Notifications
* Issues tracking
* Aggregating crash reports automatically

**Comment by Revertron** [comment]
Done.

**Comment by ameshkov** [comment]
@Revertron still waiting for the privacy policy update.

**Comment by Revertron** [comment]
I've collected the data:
> Brand and device name, OS version and kernel build, rooted status, unidentifyable android ID,
> total and free amounts of RAM, screen orientation, screen resolution and density,
> application name (AdGuard) and version.

@vbagirov Will insert them in privacy policy.

**Comment by ameshkov** [comment]
And the crash information as well.

@vbagirov please explain that the sending automatic crash reports should be explicitly allowed by the user.

**Comment by vbagirov** [comment]
@ameshkov @Revertron privacy policy has been updated.

**Comment by Revertron** [comment]
Implemented and merged.

Testing instructions: no need, we are already getting error logs automatically.

**Comment by ameshkov** [comment]
@admitrevskiy one more thing we should do -- add a low-level setting `pref.crash.reporting.auto`.
